lib/widgets/molecules/recommended_mentors_filters.dart

```diff
@@ -0,0 +1,253 @@
+
+  @override
+  State<StatefulWidget> createState() => _ExpertiseState();
+}
+
+class _ExpertiseState extends State<_Expertise> {
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final AppLocalizations l10n = AppLocalizations.of(context)!;
+
+    var skillButtons = [];
+    for (int i = 0; i < ExploreCardFiltersProvider.skills.length; i++) {
+      var skill = ExploreCardFiltersProvider.skills[i];
+      var isSelected = widget.skills.contains(skill);
+
+      skillButtons.add(OutlinedButton(
+        onPressed: () {
+          setState(() {
+            if (isSelected) {
+              widget.skills.remove(skill);
+            } else {
+              widget.skills.add(skill);
+            }
+          });
+        },
+        style: ButtonStyles.primaryRoundedRectangleButton(context).copyWith(
+          backgroundColor: MaterialStatePropertyAll(
+              (isSelected) ? theme.colorScheme.onInverseSurface : Colors.white),
+        ),
+        child: Text(l10n.exploreSearchFilterSkills(skill),
+            style: const TextStyle(color: Colors.black)),
+      ));
+      if (i != ExploreCardFiltersProvider.skills.length - 1) {
+        skillButtons.add(const SizedBox(width: 4));
+      }
+    }
+
+    bool allSkillsSelected =
+        setEquals(widget.skills, ExploreCardFiltersProvider.skills.toSet());
+
+    return Row(
+      children: [
+        Expanded(
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Text(l10n.exploreSearchFilterExpertise),
+              const SizedBox(height: 4),
+              SingleChildScrollView(
+                scrollDirection: Axis.horizontal,
+                child: IntrinsicHeight(
+                  child: Row(
+                    children: [
+                      OutlinedButton(
+                        onPressed: () {
+                          setState(() {
+                            if (allSkillsSelected) {
+                              widget.skills.clear();
+                            } else {
+                              widget.skills.clear();
+                              widget.skills
+                                  .addAll(ExploreCardFiltersProvider.skills);
+                            }
+                          });
+                        },
+                        style:
+                            ButtonStyles.primaryRoundedRectangleButton(context)
+                                .copyWith(
+                          backgroundColor: MaterialStatePropertyAll(
+                            (allSkillsSelected)
+                                ? theme.colorScheme.onInverseSurface
+                                : Colors.white,
+                          ),
+                        ),
+                        child: Text(l10n.exploreSearchFilterAll,
+                            style: const TextStyle(color: Colors.black)),
```

Comment:
Please replace all color constants with with a color role from the theme. When reviewing I only pointed out a few of these, but this convention applies to all uses of color throughout the app.

Reply:
Fixed.